extensions/amp-ad/validator-amp-ad.protoascii

```diff
@@ -77,6 +83,40 @@ tags: {  # <amp-ad>
     supported_layouts: RESPONSIVE
   }
 }
+tags: {  # <amp-ad type="custom">
+  html_format: AMP  # Ads are not allowed inside ads
+  tag_name: "AMP-AD"
+  spec_name: "amp-ad with type=custom"
+  requires_extension: "amp-ad"  # See note at top of file
+  also_requires_tag_warning: "amp-ad extension .js script"
+  # If modifying disallowed_ancestors, then please also edit
+  # extensions/amp-auto-ads/*/placement.js
+  disallowed_ancestor: "AMP-APP-BANNER"
+  attrs: {
+    name: "data-uri"
+    mandatory: true
+    value_url: {
+      protocol: "https"
+    }
+  }
+  attrs: {
+    name: "type"
+    mandatory: true
+    value: "custom"
```

Comment:
add:

```dispatch_key: NAME_VALUE_DISPATCH```

The dispatch_key will cause any AMP-AD tags with an attribute of `type=custom` to **only** be matched against this one tagspec.

Reply:
Done.